lib/Sema/CSDiag.cpp

```diff
@@ -3461,11 +3461,11 @@ bool FailureDiagnosis::visitApplyExpr(ApplyExpr *callExpr) {
     //   case (let (_, _, _)) + 1: break
     // }
     if (callExpr->isImplicit() && overloadName == "~=") {
-      auto *locator =
-          CS.getConstraintLocator(callExpr,
-                                  {ConstraintLocator::ApplyArgument,
-                                   LocatorPathElt::ApplyArgToParam(0, 0)},
-                                  /*summaryFlags=*/0);
+      auto *locator = CS.getConstraintLocator(
+          callExpr,
+          {ConstraintLocator::ApplyArgument,
+           LocatorPathElt::ApplyArgToParam(0, 0, ParameterTypeFlags())},
```

Comment:
Yeah, that's kind of unfortunate, good catch. I think a FIXME is about the best we can do here for now since, as you said, we're not relying on the flags being correct yet.

Reply:
I believe if that's a single possible overload ~= there you can get it from `CalleeCandidateInfo` and extract flags for the first parameter. Same in the other situation in CSDiagnostics - there you can definitely do that because there is a complete solution.

Reply:
Thanks for the tip! I used the `CalleeCandidateInfo` in this case which seems to work pretty well. I used the `FunctionArgApplyInfo` in the CSDiagnostics case because it simplified things a bit. Hopefully this is an ok way to do things!

Reply:
Yeah, that looks good! I'd prefer extracting it into a method on `FunctionArgApplyInfo` something like `TypeParameterFlags getParamFlags(unsigned) const`